packages/gatsby-transformer-code-examples/gatsby-node.js

```diff
@@ -0,0 +1,74 @@
+const createSchemaCustomization = ({ actions, schema }) => {
+  const { createTypes } = actions;
+
+  const typeDefs = [
+    schema.buildObjectType({
+      name: 'CodeExample',
+      fields: {
+        name: 'String!',
+        extension: 'String!',
+        absolutePath: 'String!',
+        content: 'String!',
+      },
+      interfaces: ['Node'],
+      extensions: {
+        infer: false,
+      },
+    }),
+  ];
+
+  createTypes(typeDefs);
+};
+
+async function onCreateNode({
+  node,
+  actions,
+  loadNodeContent,
+  createNodeId,
+  createContentDigest,
+}) {
+  const validExtensions = [
+    'application/javascript',
+    'text/x-java-source',
+    'application/octet-stream',
+    'application/json',
+    'application/x-httpd-php',
+    'application/x-sh',
+    'video/mp2t',
+    'text/yaml',
+  ];
+  if (!validExtensions.includes(node.internal.mediaType)) return;
+
+  const { createNode, createParentChildLink } = actions;
+
+  function transformObject(obj, id, type) {
+    const codeSampleNode = {
+      ...obj,
+      id,
+      children: [],
+      parent: node.id,
+      internal: {
+        contentDigest: createContentDigest(obj),
+        type,
+      },
+    };
+    createNode(codeSampleNode);
+    createParentChildLink({ parent: node, child: codeSampleNode });
+  }
+
+  const content = await loadNodeContent(node);
+
+  transformObject(
+    {
+      name: node.name,
+      extension: node.extension,
+      absolutePath: node.absolutePath,
+      content,
+    },
+    createNodeId(`${node.id} >>> CodeExample`),
+    'CodeExample'
+  );
+}
```

Comment:
```suggestion
  const content = await loadNodeContent(node);
  const nodeId = createNodeId(`${node.id} >>> CodeExample`);
  const codeExampleProps = {
    name: node.name,
    extension: node.extension,
    absolutePath: node.absolutePath,
    content,
  };
  const codeExampleNode = {
    ...codeExampleProps,
    id: nodeId,
    children: [],
    parent: node.id,
    internal: {
      contentDigest: createContentDigest(codeExampleProps),
      type: 'CodeExample',
    },
  };
  createNode(codeExampleNode);
  createParentChildLink({ parent: node, child: codeExampleNode });
}
```

Reply:
I see, for me, a function should ideally not exceed 4 lines, the reason is because in my experience, those kinds of functions are very readable, and they tend to encourage clean code, which I try to practice.

So I would rather do something like

```
async function onCreateNode() {
  if (!validMediaTypes()) return;
  ...
  transformObject();
}
```

Reply:
> for me, a function should ideally not exceed 4 lines

Not sure what you mean here. As long as the function logic is readable, a function can be short or long.
Extracting logic into multiple functions is not always a good thing if not necessary, as it just introduces unnecessary indirection.

Declaring a function closure within the main function, to just use it within the main function, does not help much, it's just unnecessary cognitive complexity.

Reply:
I'm sorry but I disagree.